extensions/sql-migration/src/dialog/generic/multiStepStatusDialog.ts

```diff
@@ -0,0 +1,342 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../../constants/strings';
+import { EOL } from 'os';
+import { IconPathHelper } from '../../constants/iconPathHelper';
+
+const DialogName = 'MultiStepStatusDialog';
+
+enum MultiStepResultIndex {
+	message = 0,
+	icon = 1,
+	status = 2,
+	errors = 3,
+	state = 4,
+}
+
+export enum MultiStepState {
```

Comment:
if these exact statuses already exist for something else (e.g. migration status, not sure if they line up 100% though) could we maybe reuse that instead?